src/data/list/defs.lean

```diff
@@ -524,6 +529,18 @@ def mmap_filter {m : Type → Type v} [monad m] {α β} (f : α → m (option β
 | (h :: t) := do b ← f h, t' ← t.mmap_filter, return $
   match b with none := t' | (some x) := x::t' end
 
+/--
+`mmap_diag f l` calls `f` on all elements in the "upper diagonal" of `l × l`.
```

Comment:
I think "upper diagonal" would refer to only `f 1 2, f 2 3`. Better to say `in the "upper triangular part" of`.

Reply:
And maybe as a consequence needs a rename.